sky/data/storage_utils.py

```diff
@@ -279,12 +279,18 @@ def _store_symlink(zipf, path: str, is_dir: bool):
                     zipf.write(item)
                 elif os.path.isdir(item):
                     for root, dirs, files in os.walk(item, followlinks=False):
+                        # Filter out directories in the excluded_files list
+                        # This prevents os.walk from traversing into excluded
+                        # directories
+                        dirs[:] = [
+                            d for d in dirs
+                            if os.path.join(root, d) not in excluded_files
+                        ]
+
                         # Store directory entries (important for empty
                         # directories)
                         for dir_name in dirs:
                             dir_path = os.path.join(root, dir_name)
-                            if dir_path in excluded_files:
-                                continue
```

Comment:
The deep continue is really mind-twisting, I can hard tell the semantic change without expanding the diff context 😭

Reply:
Oh, I still misunderstood the PR, finally got it with the comments